external-crates/move/crates/move-analyzer/src/symbols/runner.rs

```diff
@@ -77,8 +97,34 @@ impl SymbolicatorRunner {
                                 Some(starting_paths)
                             }
                             RunnerState::Wait => {
-                                // wait for next request
-                                symbolicate = cvar.wait(symbolicate).unwrap();
+                                // wait for next request, with timeout for parent process monitoring
+                                if parent_process_id.is_some() {
```

Comment:
nit: Why not `if let Some(parent_proc_id) = parent_process_id {` here and then you don't need the unwrap below?